xbmc/filesystem/MusicDatabaseDirectory/DirectoryNodeAlbum.cpp

```diff
@@ -22,13 +22,28 @@ CDirectoryNodeAlbum::CDirectoryNodeAlbum(const std::string& strName, CDirectoryN
 
 NODE_TYPE CDirectoryNodeAlbum::GetChildType() const
 {
+  if (GetName() == "-1")
+    return NODE_TYPE_SONG;
+
+  //! @todo: This makes all multi-disc albums album > disc > song, make disc level optional
+  CMusicDatabase musicdatabase;
```

Comment:
Add a new music lib setting to enable(default)/disable navigation from multi-disc albums to discs rather than directly to songs, and use it *here* to allow albums > discs navigation to be optional. That way we can give users original behaviour if they want it.

A search for `musiclibrary.showcompilationartists` may help you find example of how to do a new setting

Reply:
OK, I'll give it a go.

Reply:
Done that.  Default is false and navigation is as it was.  Toggling it on enables disc type navigation.  Incidentally, extraart still works in original navigation if you have set any up, so the covers change as you scroll up and down the tracks between the discs.

Reply:
Users won't see the new grouping as discs feature that way around. Make default having a "discs" level, and they can change the setting to get old behaviour back

Reply:
Done.  Not 100% sure that it should be turned off for boxsets at all, given that's where we started with all this, but as I can't see a particularly clean way of doing that I'll go with it.

Reply:
Humm, yes when viewing boxed sets you do always want album > disc...
If I think of a way to do that nicely I'll let you know. At least on by default means users with boxed sets will see the immediate advantage.